apps/next-app/src/app/products/page.tsx

```diff
@@ -182,63 +228,94 @@ export default function ProductsPage() {
+                  return (
+                    <div key={item.filterId} className='mb-12'>
+                      <FilterItem
+                        filterItems={item}
+                        onClickSort={onClickFilter}
+                      ></FilterItem>
+                    </div>
+                  );
+                })}
+              </div>
+              <div className='bg-white flex md:hidden justify-between py-5 font-medium text-sm/4 border-b'>
+                <SortMobile
+                  sortItems={filterData.sortItems}
+                  onClickApplySort={onClickSort}
+                />
+                <div className='flex divide-x'>
+                  <div className='px-3'>
+                    <ButtonChangeLayout
+                      onChangePageLayout={onChangePageLayout}
+                    />
+                  </div>
+                  <div className='px-3'>
+                    <FilterMobile
+                      filterItems={filterData.filterItems}
+                      filterTitle='Bộ Lọc'
+                      clearFilterTitle='Xóa bộ lọc'
+                      applyFilterTitle='Áp dụng bộ lọc'
+                      onClickApplyFilter={onClickFilter}
+                    />
+                  </div>
+                </div>
+              </div>
             </div>
             <div className='col-span-12 md:!col-span-9'>
-              <PagingComponent
-                productItems={productItems}
-                onClickItem={onClickItem}
-                paginationPage={paginationPage}
-                handlePageClick={handlePageClick}
-                classItem={showLayoutList ? 'flex flex-row' : 'flex flex-col'}
-                classImg={
-                  showLayoutList
-                    ? 'basis-2/5 object-contain'
-                    : 'bais-full object-contain'
-                }
-                classContent={showLayoutList ? 'basis-3/5' : 'bais-full'}
-                classPaging={
-                  showLayoutList
-                    ? 'grid gap-3 grid-cols-1'
-                    : 'grid gap-3 grid-cols-2 md:!grid-cols-4'
-                }
-                previousLabel={'Trước'}
-                nextLabel={'Sau'}
-              />
+              {showLayoutList ? (
+                <PagingListComponent
+                  productItems={productItems}
+                  paginationPage={paginationPage}
+                  previousLabel={'Trước'}
+                  nextLabel={'Sau'}
+                  onClickItem={onClickItem}
+                  handlePageClick={handlePageClick}
+                />
+              ) : (
+                <PagingGridComponent
+                  productItems={productItems}
+                  paginationPage={paginationPage}
+                  previousLabel={'Trước'}
+                  nextLabel={'Sau'}
+                  onClickItem={onClickItem}
+                  handlePageClick={handlePageClick}
+                />
+              )}
```

Comment:
```suggestion
              <ProductsList layout={showLayoutList ? 'list' : 'grid'>
                  {productItems.map(product => <ProductCard product={product} />}
              </ProductsList>
              <PaginationComponent 
                  paginationPage={paginationPage}
                  previousLabel={'Trước'}
                  nextLabel={'Sau'}
                  onClickItem={onClickItem}
                  handlePageClick={handlePageClick}
              />
```
The components look cleaner and maintenance will be completely easier.
For example, If we have problem with ProductCard or need to fix ProductCard => go to ProductCard to fix or create a new one. if the grid not showing correctly => go to ProductsList only to fix it. The same with PaginationComponent

Reply:
<PaginationComponent/> design as specify for this page only.  If you need flatten then you can copy code in this but I recommend you shouldn;t flatten something like this because if someone resuse it, they will copy and code duplicate so much.

By the way, ProductList you recommend look good but it's self just content div with classname, it's self is abuse component

Reply:
<PagingComponent
                productItems={productItems}
                onClickItem={onClickItem}
                paginationPage={paginationPage}
                handlePageClick={handlePageClick}
                display={showLayoutList ? "list" : "grid"}
                previousLabel={'Trước'}
                nextLabel={'Sau'}
              />